ext/semian/types.h

```diff
@@ -38,4 +40,28 @@ typedef struct {
   long wait_time;
 } semian_resource_t;
 
+// Internal simple integer structure
+typedef struct {
+  uint64_t key;
+  int sem_id;
+} semian_simple_integer_t;
+
+// Shared simple sliding window structure
+typedef struct {
+  int max_size;
+  int length;
+  int start;
+  int data[SLIDING_WINDOW_MAX_SIZE];
+} semian_simple_sliding_window_shared_t;
```

Comment:
The size of this structure is larger than `SHM_DEFAULT_SIZE` (and also larger than the default `PAGE_SIZE` of 4k).

Reply:
Good point. I changed the maximum size to `1,000` (which is plenty) and wrote a test to ensure that the entire shared memory structure is less than or equal to a page.